pawno/include/PPC_PlayerCommands.inc

```diff
@@ -2369,30 +2406,40 @@ COMMAND:engine(playerid, params[])
 	// Check if the player has logged in
 	if (APlayerData[playerid][LoggedIn] == true)
 	{
-		// Check if the player's admin-level is at least 1
-		if (APlayerData[playerid][PlayerLevel] >= 1)
+		//check if the player its the driver of the car, don't let the passanger to control the engine
+		if(GetPlayerState(playerid) == PLAYER_STATE_DRIVER)
 		{
 			// Get the player's vehicle
 			vehicleid = GetPlayerVehicleID(playerid);
 
-			// Get the current status of the vehicle
-			GetVehicleParamsEx(vehicleid, engine, lights, alarm, doors, bonnet, boot, objective);
-
 			// Check if the player is inside a vehicle
 			if (vehicleid != 0)
 			{
-				if (sscanf(params, "i", engine)) SendClientMessage(playerid, 0xFF0000AA, "Usage: \"/engine <EngineStatus (0/1)>\"");
+				// Get the current status of the vehicle
+				GetVehicleParamsEx(vehicleid, engine, lights, alarm, doors, bonnet, boot, objective);
+
+				//Check if the engine is running or not
+				if(engine)
```

Comment:
You can use this variable to set the state along with little maths. This will reduce the repition of coding lines.

```pawn
SetVehicleParamsEx(vehicleid, (1 - engine), lights, alarm, doors, bonnet, boot, objective);
```
see http://forum.sa-mp.com/showpost.php?p=1079313